doc/source/modulefile.rst

```diff
@@ -97,344 +95,306 @@ the *modulefile* is being loaded.
- **unsetenv**. If the environment *variable* had been defined it will
- be overwritten while loading the *modulefile*. A subsequent **unload**
+ environment *variable* like the :modulefilecmd:`setenv` command. An environment change
+ made this way will only affect the module parsing process. The :modulefilecmd:`setenv`
+ command is also useful for changing the environment prior to the ``exec``
+ or :modulefilecmd:`system` command. When a *modulefile* is unloaded, :modulefilecmd:`setenv` becomes
+ :modulefilecmd:`unsetenv`. If the environment *variable* had been defined it will
+ be overwritten while loading the *modulefile*. A subsequent :modulefilecmd:`unload`
  will unset the environment *variable* - the previous value cannot be
  restored! (Unless you handle it explicitly ... see below.)
 
-.. _unsetenv:
-
-**unsetenv** variable [value]
+.. modulefilecmd:: unsetenv variable [value]
 
  Unsets environment *variable*. However, if there is an optional *value*,
  then when unloading a module, it will set *variable* to *value*. The
- **unsetenv** command changes the process' environment like **setenv**.
+ :modulefilecmd:`unsetenv` command changes the process' environment like :modulefilecmd:`setenv`.
 
-.. _getenv:
+.. modulefilecmd:: getenv variable [value]
 
-**getenv** variable [value]
-
- Returns value of environment *variable*. If *variable* is not defined *value*
- is returned if set *_UNDEFINED_* is returned elsewhere. **getenv** command
- should be preferred over Tcl global variable **env** to query environment
+ Returns value of environment *variable*. If *variable* is not defined, *value*
+ is returned if set, ``_UNDEFINED_`` is returned otherwise. The :modulefilecmd:`getenv` command
+ should be preferred over the Tcl global variable ``env`` to query environment
  variables.
 
-.. _append-path:
-
-**append-path** [-d C|--delim C|--delim=C] [--duplicates] variable value...
+.. modulefilecmd:: append-path [-d C|--delim C|--delim=C] [--duplicates] variable value...
 
- See **prepend-path**.
+ See :modulefilecmd:`prepend-path`.
 
-.. _prepend-path:
-
-**prepend-path** [-d C|--delim C|--delim=C] [--duplicates] variable value...
+.. modulefilecmd:: prepend-path [-d C|--delim C|--delim=C] [--duplicates] variable value...
 
  Append or prepend *value* to environment *variable*. The
  *variable* is a colon, or *delimiter*, separated list such as
  ``PATH=directory:directory:directory``. The default delimiter is a colon
- ':', but an arbitrary one can be given by the *--delim* option. For
+ ':', but an arbitrary one can be given by the :option:`--delim` option. For
  example a space can be used instead (which will need to be handled in
- the Tcl specially by enclosing it in " " or { }). A space, however,
- can not be specified by the *--delim=C* form.
+ the Tcl specially by enclosing it in ``" "`` or ``{ }``). A space, however,
+ can not be specified by the :option:`--delim=C` form.
```

Comment:
`--delim`, `--duplicate` and `--index` are not declared as options so a warning is obtained when building doc. May just enclose them in double back-ticks.

Reply:
Done, though I don't like replacing semantic markup with syntactical markup again.